app/src/main/java/org/oppia/android/app/player/exploration/ExplorationFragmentPresenter.kt

```diff
@@ -56,18 +70,47 @@ class ExplorationFragmentPresenter @Inject constructor(
   fun handleViewCreated() {
     val profileDataProvider = profileManagementController.getProfile(retrieveArguments().profileId)
     profileDataProvider.toLiveData().observe(
-      fragment,
-      { result ->
-        val readingTextSize = retrieveArguments().readingTextSize
-        if (result is AsyncResult.Success && result.value.readingTextSize != readingTextSize) {
-          selectNewReadingTextSize(result.value.readingTextSize)
-
-          // Since text views are based on sp for sizing, the activity needs to be recreated so that
-          // sp can be correctly recomputed.
-          fragment.requireActivity().recreate()
+      fragment
+    ) { result ->
+      val readingTextSize = retrieveArguments().readingTextSize
+      if (result is AsyncResult.Success && result.value.readingTextSize != readingTextSize) {
+        selectNewReadingTextSize(result.value.readingTextSize)
+
+        // Since text views are based on sp for sizing, the activity needs to be recreated so that
+        // sp can be correctly recomputed.
+        fragment.requireActivity().recreate()
+      } else {
+        if (result is AsyncResult.Success) {
+          showSpotlights(result.value.numberOfLogins)
         }
       }
```

Comment:
```suggestion
      if (result is AsyncResult.Success) {
        if (result.value.readingTextSize != readingTextSize) {
          // Since text views are based on sp for sizing, the activity needs to be recreated so that
          // sp can be correctly recomputed.
          selectNewReadingTextSize(result.value.readingTextSize)
          fragment.requireActivity().recreate()
        } else showSpotlights(result.value.numberOfLogins)
      }
```

Can be condensed.

Reply:
done